pandas/core/groupby.py

```diff
@@ -2032,6 +2039,38 @@ def _get_group_keys(self):
                                           self.levels,
                                           self.labels)
 
+    def _cython_apply(self, ftype, data, axis, **kwargs):
```

Comment:
I think all this should pass down to `_cython_transform`?

Reply:
I was a little torn over this. The call signature doesn't play all that nicely with the other methods utilizing `_cython_transform`. Specifically, the `is_numeric` and `is_datetimelike` arguments aren't applicable and there are some conditionals that need to be added to prevent accidental casting of data during the fill. On top of that, we'd have to wrap the transformation to be sure to include the grouped columns as none of the other transformations do that.

Touched on this with @jreback in the comments of #19481 - there for sure needs to be a pretty comprehensive refactoring on `groupby.py` to more effectively dispatch Cython operations. For the time being I thought this approach would be the "best of the worst" option, but can take another look at wiring into transform if we feel that is a show stopper

Reply:
I don't disagree the whole thing needs some refactoring, xref also to #19354 - I'll think about it a bit more too